Lib/imaplib.py

```diff
@@ -310,17 +316,68 @@ def open(self, host='', port=IMAP4_PORT, timeout=None):
         self.host = host
         self.port = port
         self.sock = self._create_socket(timeout)
-        self.file = self.sock.makefile('rb')
+        self._file = self.sock.makefile('rb')
+
+
+    @property
+    def file(self):
+        # The old 'file' attribute is no longer used now that we do our own
+        # read() and readline() buffering, with which it conflicts.
+        # As an undocumented interface, it should never have been accessed by
+        # external code, and therefore does not warrant deprecation.
+        # Nevertheless, we provide this property for now, to avoid suddenly
+        # breaking any code in the wild that might have been using it in a
+        # harmless way.
+        warnings.warn(
+            'IMAP4.file is unsupported, can cause errors, and may be removed.',
+            RuntimeWarning)
```

Comment:
The approach is fine and it's generally what we do, but if we want to be entirely backward-compatible, a property setter might also be possible. If you want to reduce import times, you can also use import warnings locally. And sorry, but you might need a `stacklevel=2` instead of stacklevel=1 so that users have a better warning rendering (you can check this by accessing this attribute in the REPL)

Reply:
But you still don’t want it to malfunction. Test it. :-)

Reply:
Thanks for this! I should have been clearer in my review but it was time to go to bed.